processing/src/main/java/io/druid/query/metadata/metadata/SegmentMetadataQuery.java

```diff
@@ -30,12 +31,16 @@
 import io.druid.query.spec.QuerySegmentSpec;
 import org.joda.time.Interval;
 
+import java.nio.ByteBuffer;
 import java.util.Arrays;
 import java.util.EnumSet;
+import java.util.List;
 import java.util.Map;
 
 public class SegmentMetadataQuery extends BaseQuery<SegmentAnalysis>
 {
+  public static final byte[] ANALYSIS_TYPES_CACHE_PREFIX = new byte[]{0x70};
```

Comment:
ah, in that case, there's different types for the last column includerator and the cache key is required to distinguish those types. In this case, analysis types is an enum, which is a little bit different. I think what needs to happen is that the analysis_types_cache_prefix cannot be a static constant, but should instead reflect what type of analysis type it is (you can use the enum value).